src/controller/stream-controller.js

```diff
@@ -968,16 +968,26 @@ class StreamController extends BaseStreamController {
         track.levelCodec = audioCodec;
         track.id = data.id;
       }
+
       track = tracks.video;
       if (track) {
         track.levelCodec = this.levels[this.level].videoCodec;
         track.id = data.id;
       }
+
+      track = tracks.audiovideo;
+      let audioVisualLevelCodec = 'none';
+      if (track && this.levels[this.level]) {
+        audioVisualLevelCodec =
+          `${this.levels[this.level].audioCodec || 'none'}`
+          + `-${this.levels[this.level].videoCodec || 'none'}`;
+        track.id = data.id;
+      }
       this.hls.trigger(Event.BUFFER_CODECS, tracks);
       // loop through tracks that are going to be provided to bufferController
       for (trackName in tracks) {
         track = tracks[trackName];
-        logger.log(`main track:${trackName},container:${track.container},codecs[level/parsed]=[${track.levelCodec}/${track.codec}]`);
+        logger["logger"].log(`main track:${trackName},container:${track.container},codecs[level/parsed]=[${track.levelCodec || audioVisualLevelCodec}/${track.codec}]`);
```

Comment:
Something like this would do the trick and we can leave `undefined` alone:
```suggestion
        const levelCodec = trackName === 'audiovideo' ? this.levels[this.level].attrs.CODECS : this.levels[this.level][`${trackName}Codec`];
        logger.log(`main track:${trackName},container:${track.container},codecs[level/parsed]=[${levelCodec}/${track.codec}]`);
```

Reply:
fully agreeing with the obvious comment @robwalch , for your suggestion i find the ```[`${trackName}Codec`];``` part a bit too hackish however.

also there is a lot of useful stuff here that should be in level-helpers in fact. 

why not add a method (that should be one of the `Level` class would we have one)

which role would be to extrac the codec name from the level object?

then any log line can use it conveniently alike.

Reply:
What we want logged here is the "audiovideo" codecs for the level, not the audio and video split and put back together. That is present in the level attributes: `this.levels[this.level].attrs.CODECS `.

I am waiting on a reply from @karendolan on the feedback from this review.

Reply:
Hi @robwalch, I'm confused about where hls.js applies *attr.CODECS* to the *track.audiovideo*. The following, in the mp4demuxer.js, is where the audiovideo gets the 3 attributes: container, codec, and initSegment. 
```
      if (initData.audio && initData.video) {
        tracks.audiovideo = { container: 'video/mp4', codec: audioCodec + ',' + videoCodec, initSegment: duration ? initSegment : null };
      } else {
        if (initData.audio) {
          tracks.audio = { container: 'audio/mp4', codec: audioCodec, initSegment: duration ? initSegment : null };
        }

        if (initData.video) {
          tracks.video = { container: 'video/mp4', codec: videoCodec, initSegment: duration ? initSegment : null };
        }
      }
```

<img width="511" alt="Screen Shot 2020-10-09 at 1 37 17 PM" src="https://user-images.githubusercontent.com/1331728/95614565-b5d2cb80-0a34-11eb-83cc-502767cbfbf6.png">

Reply:
https://github.com/video-dev/hls.js/blob/42ccf7cac3b5ea0b2a51bdbba59b65a6780d5427/src/demux/mp4demuxer.js#L34-L56